src/app/collective-rewards/leaderboard/hooks/useGetBuildersRewards.ts

```diff
@@ -1,71 +1,230 @@
+import { useGetBackersClaimedRewardsAmount } from '@/app/collective-rewards/rewards/backers/hooks/useGetBackersClaimedRewardsAmount'
+import {
+  useGetBuildersRewardPercentage,
+  useGetGaugesBackerRewardsClaimed,
+  useGetGaugesRewardShares,
+  useGetRewardsCoinbase,
+  useGetRewardsERC20,
+  useGetTotalAllocations,
+} from '@/app/collective-rewards/rewards'
+import { getPreviousCycle } from '@/app/collective-rewards/utils/getPreviousCycle'
+import { useGetFilteredBuilders } from '@/app/collective-rewards/whitelist'
 import { formatBalanceToHuman } from '@/app/user/Balances/balanceUtils'
+import { RBTC, RIF, USD } from '@/lib/constants'
 import { usePricesContext } from '@/shared/context/PricesContext'
-import { useGetTokenProjectedReward, useGetRewardDistributedLogs } from '@/app/collective-rewards/rewards'
-import { Address, isAddressEqual } from 'viem'
-import { getLastCycleRewards } from '@/app/collective-rewards/utils/getLastCycleRewards'
-import { useBuilderContext } from '@/app/collective-rewards/user'
-import { BuilderStatusActive } from '@/app/collective-rewards/types'
 
-export const useGetBuildersRewards = (rewardToken: Address, rewardTokenSymbol?: string, currency = 'USD') => {
-  const { data: builders, isLoading: buildersLoading, error: buildersError } = useBuilderContext()
+export const useGetBuildersRewards = () => {
+  const {
+    data: whitelistedBuilders,
+    isLoading: whitelistedBuildersLoading,
+    error: whitelistedBuildersError,
+  } = useGetFilteredBuilders({ builderName: '', status: 'Active' })
+
+  const buildersAddress = whitelistedBuilders?.map(({ address }) => address)
+  const {
+    data: buildersRewardsPercentage,
+    isLoading: rewardsPercentageLoading,
+    error: rewardsPercentageError,
+  } = useGetBuildersRewardPercentage(buildersAddress)
+
+  const gauges = whitelistedBuilders?.map(({ gauge }) => gauge)
+  const {
+    data: { allocationsPercentage },
+    isLoading: allocationsLoading,
+    error: allocationsError,
+  } = useGetTotalAllocations(gauges)
+
+  const {
+    data: rewardShares,
+    isLoading: rewardSharesLoading,
+    error: rewardSharesError,
+  } = useGetGaugesRewardShares(gauges)
 
   const {
-    data: rewardDistributedLogs,
+    data: rewardsERC20,
+    isLoading: rewardsERC20Loading,
+    error: rewardsERC20Error,
+  } = useGetRewardsERC20()
+
+  const {
+    data: rewardsCoinbase,
+    isLoading: rewardsCoinbaseLoading,
+    error: rewardsCoinbaseError,
+  } = useGetRewardsCoinbase()
+
+  const { cycleStartTimestamp, cycleEndTimestamp } = getPreviousCycle()
```

Comment:
There is a mix on this, this method is depecrated and needs to be removed, the information from the epochs its under the CycleContext.